binder/README.md

```diff
@@ -0,0 +1,32 @@
+# Binder support for PEP-622 examples
+
+This directory offers [Binder](https://mybinder.org) support for these [PEP-622](https://www.python.org/dev/peps/pep-0622) examples, which depend on a currently experimental build of Python 3.10.  It is intended to facilitate testing and discussion of this proposed change to Python so that users can play with the proposed syntax without having to build and install everything themselves.
+
+This directory contains all the necessary tools for Binder support, all other content related to PEP-622 is in the rest of the repo and doesn't need binder.  This repo uses two binder tools for setup:
+
+1. An `apt.txt` file listing useful packages needed for the build.
+2. A `postBuild` script that builds/installs CPython 3.10 from the right branch and then updates that installation with some necessary dependencies for testing/running the examples.
+
+
+## Build notes
+
+Since testing the pattern matching features requires a custom Python build, we start from a standard Binder image and then build Python 3.10 in the container. Python 3.10 is new enough that we can't yet have a 3.10-based kernel for Jupyter, as Pyzmq (as of Jun 29, 2020) doesn't yet build on it. But we can still use 3.10 at the command-line to test the examples, and take advantage of the JupyterLab UI for easy experimentation (terminals, text editors, etc).
+
+The [experimental build of CPython 3.10](https://github.com/brandtbucher/cpython) (branch `patma`) needed for this is installed in `$HOME/.local/bin`, and the source of the build is available in `/tmp/cpython`, in case the user wants to experiment with it.
+
+The build dependencies listed in apt.txt can be found by using the `apt-rdepends` package:
+
+```
+apt-rdepends --build-depends --follow=DEPENDS python3-defaults | grep Build-Depends | awk -F ' +' '{print $3}'
+```
+
+Normally this container will run automatically and provide a usable URL with Jupyter in it if `jupyter-repo2docker .` is run at the top of the repository. But if anything goes wrong and you need to debug, you can find the image ID of the repo with `docker images` and then run it interactively (with root privileges) by using:
```

Comment:
Should we add a short sentence like "To work on or debug the container image generation that this directory configures use the [repo2docker](https://repo2docker.readthedocs.io/) tool locally. This is what mybinder.org uses but by running it locally you speed up the build, debug, build, debug cycles."? I use and encourage people to use the pattern of local repo2docker runs for debugging but this paragraph still came "out of the blue" for me. So I thought a little intro/motivation would be good for those who don't spend every waking hour with mybinder.org :D

Reply:
Do you mean whether I should change the README above, or whether that should be added to the r2d/binder docs?

Reply:
Was thinking of this README

Reply:
Ah, understood. Merged now, but in the long run it might have more impact to update a bit our binder/r2d docs with these patterns. I ended up getting quite a bit of 'private help' from @choldgraf on Slack on this, though I'd read a fair bit of the docs already (and it's not my first time building slightly more complex binders with custom compiled dependencies).  I'll sync with Chris/Yuvi/Erik/etc on that and will try to contribute to the docs themselves.

Thanks so much for pitching in, and more importantly - for all you've done to make Binder the incredible resource it is today!!